util/udma_barrier.h

```diff
@@ -96,6 +96,8 @@
 #define udma_to_device_barrier() asm volatile("dsb st" ::: "memory");
 #elif defined(__sparc__) || defined(__s390x__)
 #define udma_to_device_barrier() asm volatile("" ::: "memory")
+#elif defined(__loongarch__)
+#define udma_to_device_barrier() asm volatile("dbar 0" ::: "memory");
```

Comment:
Uh-oh. The macros should not have trailing semicolon in its expansion; this was neglected in previous reviews but got caught by upstream CI. You should remove the semicolons like in the previous block.

Reply:
removed, thanks.